googletest/include/gtest/internal/gtest-internal.h

```diff
@@ -1338,17 +1353,21 @@ class FlatTuple
 
 #define GTEST_TEST_NO_THROW_(statement, fail) \
   GTEST_AMBIGUOUS_ELSE_BLOCKER_ \
-  if (::testing::internal::AlwaysTrue()) { \
+  if (::testing::internal::AdditionalMessage message = ".") { \
```

Comment:
I am missing why this change is beneficial? it appears that the change amounts to the same thing anyway. What is the advantage of adding oprator bool which evaluates to  return ::testing::internal::AlwaysTrue() and not comparing to the ::testing::internal::AlwaysTrue() directly as it was before?

Reply:
We need declare `message` in the scope of the `if`. It can't go into the body of the `if` because then we cannot access it in the `else`.